utils/yoke/src/cart.rs

```diff
@@ -0,0 +1,117 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{Yoke, Yokeable};
+use std::rc::Rc;
+use std::sync::Arc;
+
+/// A [`Cart`] is a type that is acceptable as backing storage in a [`Yoke`].
+///
+/// The essential invariant that must be maintained by implementors is that `Self::Inner` references
+/// obtained from this type via `Self::get_inner()` must continue to be valid for the
+/// duration of this type provided that this type is never accessed via `&mut` references.
```

Comment:
What is "the duration of this type"?

Reply:
complete lifecycle of this type (until it is destroyed)